sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/interface.scala

```diff
@@ -45,35 +45,28 @@ case class CatalogFunction(
  */
 case class CatalogStorageFormat(
     locationUri: Option[String],
-    inputFormat: Option[String],
-    outputFormat: Option[String],
-    serde: Option[String],
-    compressed: Boolean,
-    serdeProperties: Map[String, String]) {
+    provider: Option[String],
```

Comment:
To remove `CreateHiveTableAsSelectLogicalPlan`, we can add a `CatalogTable` member into the signature of `CreateTableUsingAsSelect`. After code changes, the signature of `CreateTableUsingAsSelect` becomes
```
    tableDesc: CatalogTable,
    provider: String,
    child: LogicalPlan,
    mode: SaveMode
```

After this PR is merged, we also can remove `provider` from the signature!